dataset/src/test/scala/frameless/functions/NonAggregateFunctionsTests.scala

```diff
@@ -0,0 +1,450 @@
+        .map(_.getAs[Int](0))
+        .collect().toList
+
+      val typedDS = TypedDataset.create(values.map(X1(_)))
+      val res = typedDS
+        .select(ascii(typedDS('a)))
+        .collect()
+        .run()
+        .toList
+
+      res ?= resCompare
+    }
+
+    check(forAll(prop _))
+  }
+
+
+  test("atan") {
+    val spark = session
+    import spark.implicits._
+
+    def prop[A: CatalystNumeric : TypedEncoder : Encoder](value: List[A]) = {
+      val cDS = session.createDataset(value)
+      val resCompare = cDS
+        .select(org.apache.spark.sql.functions.atan(cDS("value")))
+        .map(_.getAs[Double](0))
+        .map(DoubleBehaviourUtils.nanNullHandler)
+        .collect().toList
+
+
+      val typedDS = TypedDataset.create(value.map(X1(_)))
+      val res = typedDS
+        .select(atan(typedDS('a)))
+        .deserialized
+        .map(DoubleBehaviourUtils.nanNullHandler)
+        .collect()
+        .run()
+        .toList
+
+      res ?= resCompare
+    }
+
+
+    check(forAll(prop[Int] _))
+    check(forAll(prop[Long] _))
+    check(forAll(prop[Short] _))
+    check(forAll(prop[BigDecimal] _))
+    check(forAll(prop[Byte] _))
+    check(forAll(prop[Double] _))
+  }
+
+  test("asin") {
+    val spark = session
+    import spark.implicits._
+
+    def prop[A: CatalystNumeric : TypedEncoder : Encoder](value: List[A]) = {
+      val cDS = session.createDataset(value)
+      val resCompare = cDS
+        .select(org.apache.spark.sql.functions.asin(cDS("value")))
+        .map(_.getAs[Double](0))
+        .map(DoubleBehaviourUtils.nanNullHandler)
+        .collect().toList
+
+
+      val typedDS = TypedDataset.create(value.map(X1(_)))
+      val res = typedDS
+        .select(asin(typedDS('a)))
+        .deserialized
+        .map(DoubleBehaviourUtils.nanNullHandler)
+        .collect()
+        .run()
+        .toList
+
+      res ?= resCompare
+    }
+
+
+    check(forAll(prop[Int] _))
+    check(forAll(prop[Long] _))
+    check(forAll(prop[Short] _))
+    check(forAll(prop[BigDecimal] _))
+    check(forAll(prop[Byte] _))
+    check(forAll(prop[Double] _))
+  }
+
+  test("atan2") {
+    val spark = session
+    import spark.implicits._
+
+    def prop[A: CatalystNumeric : TypedEncoder : Encoder, B: CatalystNumeric : TypedEncoder : Encoder](value: List[X2[A, B]])
+            (implicit encEv: Encoder[X2[A,B]]) = {
+      val cDS = session.createDataset(value)
+      val resCompare = cDS
+        .select(org.apache.spark.sql.functions.atan2(cDS("a"), cDS("b")))
+        .map(_.getAs[Double](0))
+        .map(DoubleBehaviourUtils.nanNullHandler)
+        .collect().toList
+
+
+      val typedDS = TypedDataset.create(value)
+      val res = typedDS
+        .select(atan2(typedDS('a), typedDS('b)))
+        .deserialized
+        .map(DoubleBehaviourUtils.nanNullHandler)
+        .collect()
+        .run()
+        .toList
+
+      res ?= resCompare
+    }
+
+
+    check(forAll(prop[Int, Long] _))
```

Comment:
What about other combinations?

Reply:
there should be 6 combinations. should I test exhaustively over the numeric types?